### PR TITLE
chore(lint): update ESLint with Config-fxa

### LINKED ISSUE #2620: Update .eslintrc files to use the shared eslint-config-fxa module
We should update the current [./.eslintrc](https://github.com/mozilla/fxa-content-server/blob/master/.eslintrc) to extend the [**eslint-config-fxa**](https://github.com/pdehaan/eslint-config-fxa) module, to further having one single source of ESLint rules.

I think it should be as easy as installing **eslint-config-fxa** and tweaking the two .eslintrc files:
```json
{
  "extends": "fxa/client",

  "rules": {
    "handle-callback-err": 0,
    "strict": [2, "never"]
  }
}
```
**Listing 1:** _/.eslintrc_

---

```json
{
  "env": {
    "jquery": true
  },

  "rules": {
    "consistent-this": 0,
    "strict": [2, "function"]
  }
}
```
**Listing 2:** _/app/.eslintrc_

### PR DESCRIPTION
**Pull request by TDA**
Fixes #2620
Update the ESLint configuration to use global repo rules
@pdehaan 
@zaach

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -73,6 +73,8 @@
   "devDependencies": {
     "awsbox": "0.7.0",
     "coveralls": "2.11.1",
+    "eslint": "0.23.0",
```
Comment (pdehaan) [inline package.json]: Don't need the **eslint** dep here since we're using **grunt-eslint**.

**Comment by pdehaan** [comment]
Re: https://github.com/mozilla/fxa-content-server/pull/2597#discussion_r32463526

Not sure if we want to add the `camelcase: 2` rule to the base .eslintrc (so it gets inherited).
Or I can add it to the **eslint-config-fxa** "fxa/client" config if you prefer. But we'll still need to explicitly ignore any violations throughout the codebase where we want/need to use camelcase variables for various reasons.

**Comment by pdehaan** [comment]
I like it. My https://github.com/mozilla/fxa-content-server/pull/2631#issuecomment-114657096 (`camelcase`) was more of a issue for the other thread.
But let's remove the **eslint** dep before we merge this since it's a bit redundant.

I'll leave it for @zaach or @vladikoff or @shane-tomlinson to r+ and merge and we can address any other nits as they come up.

**Comment by TDA** [comment]
Done :)

**Comment by pdehaan** [comment]
Actually, in the spirit of Work Week Fever, I'll merge this.

**Pull request merged**